### DISCUSSION TITLE
Standardize Hybrid Mobile Components

### DISCUSSION THREAD

**Discussion by theswerd** [MobileNativeFoundation/discussions#100] [Proposals]
# Idea
React Native, Xamarin/Maui, and Flutter have a ton of overlap, yet their overlapping components have completely different names, making it harder for mobile developers to learn other frameworks, and harder for companies to experiment with moving code to other frameworks.

## Example
Text fields are a component seen in every app. Flutter calls them **TextField**, React Native calls them **TextInput**, and Xamarin calls them **Entry**. When I first searched up **TextField Xamarin nothing came up**!

## Use
Creating a standard recommended set of components, and names for these components or a unified cross cross-platform framework almanac to make translating between them much easier and help with long-term sustainability for apps.

**Comment by shivankD** [comment]
I hear you and share your concern. In a past life I worker for a company that was largely xamarin native/forms based. We ended up creating our own custom component library and standardized the names based on what made sense to us. 

We had a few principles though that we strictly followed - if we're creating a button, it should still extend the native button components and all the methods that `button` exposes should be applicable to our design system components etc. 

Reach out if you want to talk more or share ideas buddy :)
